src/zenml/integrations/mlflow/mlflow_utils.py

```diff
@@ -0,0 +1,159 @@
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import os
+from typing import Any, Callable, Optional, Type
+
+import mlflow  # type: ignore
+from mlflow import set_experiment
+from mlflow.tracking import set_tracking_uri  # type: ignore
+
+from zenml.io.utils import get_global_config_directory
+from zenml.pipelines.base_pipeline import BasePipeline
+from zenml.steps import BaseStep
+
+
+def local_mlflow_backend() -> str:
+    """Returns the local mlflow backend inside the global zenml directory"""
+    local_mlflow_backend_uri = os.path.join(
+        get_global_config_directory(), "mlruns"
+    )
+    if not os.path.exists(local_mlflow_backend_uri):
+        os.mkdir(local_mlflow_backend_uri)
+    return local_mlflow_backend_uri
+
+
+def setup_mlflow(
+    backend_store_uri: Optional[str] = None, experiment_name: str = "default"
+) -> None:
+    """Setup all mlflow related configurations. This includes specifying which
+    mlflow tracking uri should b e used and which experiment the tracking
+    will be associated with.
+
+    Args:
+        backend_store_uri: The mlflow backend to log to
+        experiment_name: The experiment name under which all runs will be
+                         tracked
+
+    """
+    # TODO [HIGH]: Implement a way to get the mlflow token and set
+    #  it as env variable at MLFLOW_TRACKING_TOKEN
+    if not backend_store_uri:
+        backend_store_uri = local_mlflow_backend()
+
+    os.environ["MLFLOW_TRACKING_URI"] = backend_store_uri
+    set_tracking_uri(backend_store_uri)
+    # Set which experiment is used within mlflow
+    set_experiment(experiment_name)
+
+
+def enable_mlflow_init(
+    original_init: Callable[[BasePipeline, BaseStep, Any], None],
+    experiment_name: Optional[str] = None,
+) -> Callable[..., None]:
+    """Outer decorator function for extending the __init__ method for pipelines
+    that should be run using mlflow
+
+    Args:
+        original_init: The __init__ method that should be extended
+        experiment_name: The users chosen experiment name to use for mlflow
+
+    Returns:
+        the inner decorator which extends the __init__ method
+    """
+
+    def inner_decorator(
+        self: BasePipeline, *args: BaseStep, **kwargs: Any
+    ) -> None:
+        """Inner decorator overwriting the pipeline __init__
+        Makes sure mlflow is properly set up and all mlflow logging takes place
+        within one mlflow experiment that is associated with the pipeline
+        """
+        original_init(self, *args, **kwargs)
+        setup_mlflow()
+        mlflow.set_experiment(experiment_name if experiment_name else self.name)
```

Comment:
You should pass the experiment name to the `setup_mlflow` function instead. That way you have a nice and clean separation of concerns here.